basic-usage/configuration.md

```diff
@@ -6,9 +6,83 @@ published: True
+
+`FiddlerCoreStartupSettingsBuilder` provides fluent API, for example a convenient usage is:
+```c#
+FiddlerCoreStartupSettings startupSettings =
+    new FiddlerCoreStartupSettingsBuilder()
+        .ListenOnPort(fiddlerCoreListenPort)
+        .RegisterAsSystemProxy()
+        .Build();
+
+FiddlerApplication.Startup(startupSettings);
+```
+
+The following configuration methods of `FiddlerCoreStartupSettingsBuilder` are available:
+###### Common settings:
+
+- `ListenOnPort(int)`: Specifies the port on which FiddlerCore will listen. If 0 is used, random port is assigned.
+- `AllowRemoteClients()`: Allows FiddlerCore to accept requests from outside of the current machine, e.g. remote computers and devices.
+
+>warningBe cautious when allowing remote clients to connect to FiddlerCore. If an attacker is able to proxy its traffic through this FiddlerCore instance, it could circumvent IPSec traffic rules and intranet firewalls.
+
+###### Proxy settings:
+>tipThere are a lot of possible systems and types of connections which might have to be modified in order to set proper proxy settings, and the following methods handle only the most common scenarios. For more advanced proxy configuration, see [Register as System Proxy]({%slug register-as-system-proxy %}) article.
+
+- `RegisterAsSystemProxy()`: Modifies the local LAN connection's proxy settings to point to the port on which FiddlerCore is listening on localhost. 
+- `MonitorAllConnections()`: Modifies all system connections' proxy settings to point to the port on which FiddlerCore is listening on localhost.
+- `ChainToUpstreamGateway()`: Sets the current proxy settings as an upstream gateway proxy settings in case FiddlerCore is registered as a system proxy. For example, if the application is running in a corporate environment behind a corporate proxy, the corporate proxy will be used as an upstream gateway proxy.
+>noteChaining when upstream gateway proxy settings are configured to use PAC file is supported only on Windows.
+- `SetUpstreamProxySettingsTo(ProxySettings)`: Modifies the system connection(s) proxy settings as specified in the passed instance for ProxySettings class.
+- `CaptureFTP()`: Modifies the system's proxy FTP-related settings to point to the port on which FiddlerCore is listening on localhost.
+- `HookUsingPACFile()`: Modifies current proxy settings to be configured using [PAC](https://en.wikipedia.org/wiki/Proxy_auto-config) file. On the other side, FiddlerCore serves a PAC file which is used to modify the connection. The default PAC file which is served can be configured by changing the "fiddler.proxy.pacfile.text" preference:
+```c#
+FiddlerApplication.Prefs.SetStringPref("fiddler.proxy.pacfile.text", "<pac file content>");
```

Comment:
"fiddler.proxy.pacfile.text" is only the body of the function FindProxyForURL(url, host) from the pac file.

Reply:
something like "return 'PROXY 127.0.0.1:8888';"